language/functional_tests/tests/testsuite/generics/signatures/reference_as_type_actual_for_struct.mvir

```diff
@@ -0,0 +1,10 @@
+module M {
+    struct S<T> {}
+
+    foo() {
+        let s: Self.S<&u64>;
+        return;
+    }
+}
+
+// check: INVALID_SIGNATURE_TOKEN
```

Comment:
Could we add one more test similar to this one that checks the procedure case? Something like

```
id<T>(t: T): T {
  return move(t);
}

use_id(x: u64) {
  let x_ref: &u64;
  Self.id(&x); // should be rejected 
}
```

Reply:
Already have one: `reference_as_type_actual_for_bytecode_instruction.mvir`. (The one above.) Does exactly what you suggests.